src/robot/running/arguments/typeinfo.py

```diff
@@ -276,6 +280,37 @@ def from_sequence(cls, sequence: 'tuple|list') -> 'TypeInfo':
             return infos[0]
         return cls('Union', nested=infos)
 
+    @classmethod
+    def from_variable(cls, variable: 'str|VariableMatch',
+                      handle_list_and_dict: bool = True) -> 'TypeInfo|None':
+        """Construct a ``TypeInfo`` based on a variable."""
```

Comment:
Here could be an example about the variable syntax and should be a note about this being new in RF 7.3.